syntax = "proto3";

service QueryNode {
	rpc AskQuery(Query) returns (stream Result) {}
	rpc AddDocuments(stream Document) returns (Status) {} 
	rpc DeleteDocuments(DocumentId) returns (Status) {}
	rpc FetchDocuments(DocumentId) returns (Document) {}
}

service DataNode {
	
	rpc AskQuery(Query)	returns (stream Result) {}
	rpc WriteRequest(stream Document) returns (Status) {}
	rpc WriteReply(Status) returns (Status) {}
	rpc DeleteRequest(DocumentId) returns (Status) {}
	rpc DeleteReply(Status) returns (Status) {}
	rpc FetchDocuments(DocumentId) returns (Document) {}
}

message Query
{
	string query=1;
}

message Result
{
	int32 docid=1;
	string title=2;
}

message Status
{
	string content=1;
}

message Document
{
	int32 docid=1;
	string title=2;
	string content=3;
}

message DocumentId
{
	int32 docid = 1;
}